cpp/test/io.cpp

```diff
@@ -77,6 +77,39 @@ void test_fides_function()
   writer1.write(0.0);
 }
 
+template <std::floating_point T>
```

Comment:
Just put the test in Python, then we have coverage of the code and its Python interface.

Reply:
The choice I was given was but an illusion D: